File: 005/005.rb
#!/usr/local/bin/ruby -w

# What is the smallest positive number that is
# evenly divisible by all of the numbers from 1 to 20?

def finder()
    found = false
    step = 20
    while true
        # If n % k == 0 then obviously n % (any factor of k) == 0,
        # so we can skip half the steps in this range
        (11..20).each do |i|
            if step % i == 0
                found = true
            else
                found = false
                break
            end
        end
        if found
            return step
        end
        step += 20
    end
end

puts "The number in question is #{finder()}."

File: 005/005.py
#!/usr/bin/env python

# What is the smallest positive number that is
# evenly divisible by all of the numbers from 1 to 20?


def finder():
    found = False
    step = 20
    while True:
        # If n % k == 0 then obviously n % (any factor of k) == 0,
        # so we can skip half the steps in this range
        for i in range(11, 21):
            if step % i == 0:
                found = True
            else:
                found = False
                break
        if found:
            return step
        step += 20


if __name__ == "__main__":
    number = finder()
    print("The number in question is %d." % number)
